src/main/java/org/candlepin/service/exception/user/UserMissingOwnerException.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2009 - 2023 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.service.exception.user;
+
+
+/**
+ * The UserMissingOwnerException is used to reject a user validation action.
+ */
+public class UserMissingOwnerException extends UserServiceException {
```

Comment:
Adapters had (well, still has) been using an OwnerCurator in order to look up the full details of the Owner(s) that the user is a part of, as part of the user lookup. When the OwnerCurator returns back null for "nah, there's no owner by that key". then Adapters would either throw an error or return back an empty list of accessible owners depending on the method.

I posed a question in a google meet call with Candlepin folks, including @Ceiu about Adapters use of curators. It was determined that _*any use of curators by Adapters*_ is ill-informed, no good, bad, why-is-this-happening-ow-my-eyes.

So, at some point down the line, and hopefully soon, Adapters will no longer be using the owner curator to fill in the owner details when looking up a user, and will instead return an `OwnerInfo` (though if an `OwnerKeyInfo` was a thing I'd do that) which only has info for `key`. Which means that Adapters won't be erroring on missing owners, and it'll now be on core Candlepin to determine what to do when it gets back a user who belongs to an Owner that doesn't exist. Which means that this exception probably isn't needed, but not 100% sure on that. At the very least, eventually Adapters will never throw the exception, because in order to throw it, Adapters must be able to look up the Owner, and the only way it can do that is through Curator, which Adapters should not use.